File: problems/数学与位运算/172.阶乘后的零.py
#!/usr/bin/env python3
"""
172. 阶乘后的零
"""
def trailing_zeroes(n: int) -> int:
    count = 0
    while n:
        n //= 5
        count += n
    return count

if __name__ == "__main__":
    # 测试用例
    print(trailing_zeroes(3))   # 预期: 0
    print(trailing_zeroes(5))   # 预期: 1
    print(trailing_zeroes(30))  # 预期: 7
    print(trailing_zeroes(0))   # 预期: 0

File: problems/数学与位运算/172.阶乘后的零.cpp
#include <iostream>
using namespace std;

int trailingZeroes(int n) {
    int count = 0;
    while (n) {
        n /= 5;
        count += n;
    }
    return count;
}

int main() {
    // 测试用例
    cout << trailingZeroes(3) << endl;  // 预期: 0
    cout << trailingZeroes(5) << endl;  // 预期: 1
    cout << trailingZeroes(30) << endl; // 预期: 7
    cout << trailingZeroes(0) << endl;  // 预期: 0
    return 0;
}
